### PR TITLE
Minor FPU cleanup

### PR DESCRIPTION
**Pull request by kklobe**
# Description

I wasn't able to fix the Critical Path issue (#610) yet, but I was reminded of a few things in the FPU code that annoyed me. This is a largely non-functional cleanup, but there a few small optimizations and one fix. 

I extended the single precision rounding fix to all rounding modes, instead of just `ROUND_Chop`. This is really only noticeable when logging the output, as I didn't notice any visible differences during e.g. Quake gameplay or any of the demos that heavily use single precision mode, but for consistency it should be done for all modes.

The minor optimizations are to the log-related functions.

The cleanup is mostly `std::` prefixing, some more de-Bitu-ing, and the bigger change of using `std::optional` for the array of the 64-bit FILD/FIST memcpy trick, thus eliminating the corresponding array of `bool` used to track the memcpy state.

# Manual testing

MCPDIAG gives the same result:

CleanShot 2024-07-24 at 21 47 27@2x

 as `main`:

CleanShot 2024-07-24 at 21 48 05@2x

The Quake screen size option slider still works:

CleanShot 2024-07-24 at 21 49 27@2x

I also tested:

- Sunflower
- Carmageddon
- moralhardcandy
- Heaven 7
- Quake (QTD for performance)
- multikolor
- toontown
- Z.A.R. demo (MMX mode)
- matrix (MMX demo https://www.pouet.net/prod.php?which=1100)

# Checklist

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [x] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [ ] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [x] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

File: src/fpu/fpu.cpp [hunk 1]
```diff
@@ -117,8 +117,8 @@ void FPU_ESC0_EA(Bitu rm,PhysPt addr) {
 }
 
 void FPU_ESC0_Normal(Bitu rm) {
-	Bitu group=(rm >> 3) & 7;
-	Bitu sub=(rm & 7);
+	const uint8_t group=(rm >> 3) & 7;
+    const uint8_t sub=(rm & 7);
```
Comment (johnnovak) [inline src/fpu/fpu.cpp]: indentation

File: src/fpu/fpu.cpp [hunk 2]
```diff
@@ -152,8 +152,8 @@ void FPU_ESC0_Normal(Bitu rm) {
 
 void FPU_ESC1_EA(Bitu rm,PhysPt addr) {
 // floats
-	Bitu group=(rm >> 3) & 7;
-	Bitu sub=(rm & 7);
+    const uint8_t group=(rm >> 3) & 7;
+    const uint8_t sub=(rm & 7);
```
Comment (johnnovak) [inline src/fpu/fpu.cpp]: indentation (just reformat the whole file)

File: src/fpu/fpu.cpp [hunk 3]
```diff
@@ -429,8 +429,8 @@ void FPU_ESC4_EA(Bitu rm,PhysPt addr) {
 
 void FPU_ESC4_Normal(Bitu rm) {
 	/* LOOKS LIKE number 6 without popping */
-	Bitu group=(rm >> 3) & 7;
-	Bitu sub=(rm & 7);
+    const uint8_t group=(rm >> 3) & 7;
+    const uint8_t sub=(rm & 7);
```
Comment (johnnovak) [inline src/fpu/fpu.cpp]: formatting (spaces around the `=`)

File: src/fpu/fpu_instructions.h [hunk 4]
```diff
@@ -607,8 +607,8 @@ static void FPU_FLDENV(PhysPt addr){
 
 static void FPU_FSAVE(PhysPt addr){
 	FPU_FSTENV(addr);
-	Bitu start = (cpu.code.big?28:14);
-	for(Bitu i = 0;i < 8;i++){
+    PhysPt start = (cpu.code.big?28:14);
```
Comment (johnnovak) [inline src/fpu/fpu_instructions.h]: indentation

File: include/fpu.h [hunk 5]
```diff
@@ -89,8 +91,8 @@ enum FPU_Round : uint8_t {
 struct FPU_rec {
 	FPU_Reg regs[9]         = {};
 #if !C_FPU_X86
-	int64_t regs_memcpy[9]  = {}; // for FILD/FIST 64-bit memcpy fix
-	bool use_regs_memcpy[9] = {};
+	std::optional<int64_t> regs_memcpy[9] = {}; // for FILD/FIST 64-bit
+	                                            // memcpy fix
```
Comment (johnnovak) [inline include/fpu.h:95]: Minor: can you put the comment above the line while you're at it? 😄

File: src/fpu/fpu.cpp (hunk 2 above)
Comment (kklobe) [reply src/fpu/fpu.cpp]: yeah, seems like the script doesn't quite do the right thing all the time.

**Review by johnnovak — APPROVED** [review]
Very nice @kklobe and extra points for the thorough testing 🚀

File: src/fpu/fpu.cpp (hunk 2 above)
Comment (kklobe) [reply src/fpu/fpu.cpp]: oh, wait, looks like you were looking at an outdated commit, nevermind

**Pull request merged**